spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java

```diff
@@ -256,14 +307,18 @@ public void onNext(Void aVoid) {
 
 			@Override
 			public void onError(Throwable t) {
-				terminateSpan(t);
-				this.actual.onError(t);
+				try (Scope scope = this.currentTraceContext.maybeScope(span.context())) {
```

Comment:
in brave typically, we've done like you are doing here.. decorate the onXXX hooks, but not `request` or `cancel`. 

That said there is not a huge amount of feedback for the impact of not doing  `request` or `cancel` either. Just each of these things has a chance to have a bug, so doing less means less weird unit tests to write and keep passing.